jack.py

```diff
@@ -1247,6 +1248,98 @@ def callback_wrapper(_):
+            same thread whenever the transport is rolling, or when any
+            client has requested a new position in the previous
+            cycle. The first cycle after `set_timebase_callback()` is
+            also treated as a new position, or the first cycle after
+            `jack_activate()` if the client had been inactive.
+
+            The timebase master may not use its `pos` argument to set
+            `pos.frame`. To change position, use
+            `jack_transport_reposition()` or
+            `jack_transport_locate()`. These functions are
+            realtime-safe, the `timebase_callback` can call them
+            directly.
+
+            It must have this signature::
+
+                callback(state:int,
+                         frames:int,
+                         pos:CFFI_struct,
+                         new_pos:bool) -> None
+
+            * state : int
+                current transport state.
+            * frames : int
+                number of frames in current period.
+            * pos : Position CFFI structure
+                position structure for the next cycle; `pos.frame` will
+                be its frame number. If `new_pos` is `False`, this
+                structure contains extended position information from
+                the current cycle. If `True`, it contains whatever was
+                set by the requester. The `timebase_callback`'s task is
+                to update the extended information here. See `JACK
+                transport documentation`__ for the available fields.
+            * new_pos : bool
+                `True` (non-zero) for a newly requested `pos`, or for
+                the first cycle after the `timebase_callback` is
+                defined.
+        conditional : bool
+            set to `True` for a conditional request. If set to `True`
+            this call will fail if there is already a timebase master
+
+        Returns
+        -------
+        bool
+            `True` if new timebase master was set. `False` if
+            conditional request fail bacause another master is already
+            registered.
+
+__ http://jackaudio.org/files/docs/html/structjack__position__t.html
+
+        """
+        @self._callback("JackTimebaseCallback")
+        def callback_wrapper(state, frames, pos, new_pos, _):
+            return callback(state, frames, pos, new_pos)
+
+        return_code = _lib.jack_set_timebase_callback(self._ptr,
+                                                      conditional,
+                                                      callback_wrapper,
+                                                      _ffi.NULL)
+
+        if return_code == 0:
+            return True
+        elif conditional and return_code == -1:
+            return False
+        else:
+            raise JackError("{0} ({1})"
+                            .format("Error setting timebase callback",
+                                    return_code))
```

Comment:
I think the logic here is more complicated than necessary.
And you should use `_check()`.

What about this?

```Python
err = _lib.jack_set_timebase_callback(self._ptr, conditional,
                                      callback_wrapper, _ffi.NULL)

# Because of a bug in JACK2 (see TODO), we also check for -1:
if conditional and err in (_errno.EBUSY, -1):
    return False
_check(err, "Error setting timebase callback")
return True
```

You just have to add the URL to your bug report.